status/conditions.go

```diff
@@ -1,64 +1,68 @@
 package status
 
 import (
+	"context"
 	"time"
 
+	"github.com/awslabs/operatorpkg/event"
+	"github.com/awslabs/operatorpkg/object"
 	"github.com/prometheus/client_golang/prometheus"
-	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime"
+	"github.com/samber/lo"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
 )
 
-const (
-	metricLabelName            = "name"
-	metricLabelNamespace       = "namespace"
-	metricLabelKind            = "kind"
-	metricLabelConditionType   = "type"
-	metricLabelConditionStatus = "status"
-	metricNamespace            = "operator"
-	metricSubsystem            = "status_condition"
-)
-
-type ConditionedObject interface {
-	v1.Object
-	runtime.Object
-	GetConditions() Conditions
+func NewConditions(object Object, conditions []Condition) Conditions {
+	return Conditions{object: object, conditions: lo.ToPtr(conditions)}
```

Comment:
Array should be pass-by-reference anyways so I'm unclear why you need a pointer here

Reply:
Mutability